datastore/txn.go

```diff
@@ -66,7 +66,7 @@ func NewTxnFrom(ctx context.Context, rootstore ds.TxnDatastore, id uint64, reado
 		if err != nil {
 			return nil, err
 		}
-		multistore := MultiStoreFrom(rootTxn)
+		multistore := MultiStoreFrom(ShimTxnStore{rootTxn})
```

Comment:
question: Why have you made this change?  There should be no need for it root is `IterableTxnDatastore`

Reply:
`IterableTxnDatastore` does not implement `Close` and `MultiStoreFrom` now requires a `ds.Datastore`.

Reply:
Ah okay - thanks for clarifying :)